ext/opcache/jit/zend_jit_ir.c

```diff
@@ -16644,6 +16644,126 @@ static bool zend_jit_may_be_in_reg(const zend_op_array *op_array, zend_ssa *ssa,
 	return 1;
 }
 
+static void jit_frameless_icall0(zend_jit_ctx *jit, const zend_op *opline)
+{
+	jit_SET_EX_OPLINE(jit, opline);
+
+	void *function = ZEND_FLF_HANDLER(opline);
+	zend_jit_addr res_addr = RES_ADDR();
+	ir_ref res_ref = jit_ZVAL_ADDR(jit, res_addr);
+	jit_set_Z_TYPE_INFO(jit, res_addr, IS_NULL);
+	ir_CALL_1(IR_VOID, ir_CONST_ADDR((size_t)function), res_ref);
+	zend_jit_check_exception(jit);
+}
+
+static void jit_frameless_icall1(zend_jit_ctx *jit, const zend_op *opline, uint32_t op1_info)
+{
+	jit_SET_EX_OPLINE(jit, opline);
+
+	const zend_frameless_function_info *flfi = ZEND_FLF_INFO(opline);
+	if ((op1_info & MAY_BE_RC1) && (flfi->arg_addref_info & (1 << 0))) {
+		op1_info |= MAY_BE_RCN;
+	}
+
+	void *function = ZEND_FLF_HANDLER(opline);
+	zend_jit_addr res_addr = RES_ADDR();
+	zend_jit_addr op1_addr = OP1_ADDR();
+	ir_ref res_ref = jit_ZVAL_ADDR(jit, res_addr);
+	ir_ref op1_ref = jit_ZVAL_ADDR(jit, op1_addr);
+	jit_set_Z_TYPE_INFO(jit, res_addr, IS_NULL);
+	zend_jit_zval_check_undef(jit, op1_ref, opline->op1.var, opline, 1);
+	op1_ref = jit_ZVAL_DEREF_ref(jit, op1_ref);
```

Comment:
This should be optimized to
```c
	if (op1_info & MAY_BE_UNDEF) {
		zend_jit_zval_check_undef(jit, op1_ref, opline->op1.var, opline, 1);
	}
	if (op1_info & MAY_BE_REF) {
		op1_ref = jit_ZVAL_DEREF_ref(jit, op1_ref);
	}
```
The similar changes should be done for all ``jit_frameless_call*``